p2p/discovery.py

```diff
@@ -84,6 +85,12 @@ def __init__(self, privkey: datatypes.PrivateKey, address: kademlia.Address,
         self.bootstrap_nodes = [kademlia.Node.from_uri(node) for node in bootstrap_nodes]
         self.this_node = kademlia.Node(self.pubkey, address)
         self.kademlia = kademlia.KademliaProtocol(self.this_node, wire=self)
+        self.cancel_token = CancelToken('DiscoveryProtocol')
+
+    async def lookup_random(self, cancel_token: CancelToken) -> List[kademlia.Node]:
```

Comment:
Am I correct that this is only used in the `_test` script below.  If it's only for testing can we get this out of the base class and into a subclass used only for testing?

Reply:
It is also going to be used in production code (coming on my next PR). It will be called periodically to make sure the kademlia routing table is always fresh